### PR TITLE
[Dynamo] Guard name should be valid Python identifier

### LINKED ISSUE #96149: Dynamo failure when generating guard with eval()
### 🐛 Describe the bug

Reported by Meta internal use case, but I can reproduce with the following script:
```
import torch
import torch._dynamo
from torch import nn
from torchvision import models


class RegNet1(nn.Module):

    def __init__(self):
        super().__init__()
        model = models.regnet_y_400mf()
        modules = list(model.children())
        self.model = nn.Sequential(modules[0], *modules[1][:2])

    def forward(self, x):
        return self.model(x)

m = RegNet1()
x = torch.rand([4, 3, 64, 64])
opt_m = torch._dynamo.optimize("eager")(m)
print(m(x))
print(opt_m(x))
```

Error
```
Traceback (most recent call last):
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/convert_frame.py", line 364, in _compile
    check_fn = CheckFunctionManager(
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/guards.py", line 641, in __init__
    guard.create(local_builder, global_builder)
  File "/scratch/ybliang/work/repos/pytorch/torch/_guards.py", line 158, in create
    return self.create_fn(self.source.select(local_builder, global_builder), self)
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/guards.py", line 299, in CONSTANT_MATCH
    val = self.get(guard.name)
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/guards.py", line 147, in get
    return eval(name, self.scope, CLOSURE_VARS)
  File "<string>", line 1
    getattr(self.model, '1').block1-0._has_hooks
                                      ^
SyntaxError: unexpected EOF while parsing
```

### Versions

N/A

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire @davidberard98

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #96149

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: test/dynamo/test_modules.py [hunk 1]
```diff
@@ -683,6 +692,17 @@ def forward(self, x):
         return x * self.scale
 
 
+class RegNet(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        model = torchvision_models.regnet_y_400mf()
```
Comment (yanboliang) [inline test/dynamo/test_modules.py:698]: I purposely use this complicated module architecture to test if we have a good coverage.

Comment (ezyang) [reply test/dynamo/test_modules.py:698]: Running all of resnet is pretty slow, I would prefer a minimal repro instead. To test a full model, we should see why our benchmark suite did not catch it

**Review by ezyang — APPROVED** [review]
approving because it's functionally correct, just not a fan of the testing strategy

**Review by jansel — APPROVED** [review]

File: test/dynamo/test_modules.py (hunk 1 above)
Comment (yanboliang) [reply test/dynamo/test_modules.py]: Make sense, I updated with a simplified version. Thanks.

**Pull request closed without merging**